### PR TITLE
Add Nonetype check in stochastic_ternary.__call__

### PR DESCRIPTION
**Pull request by jecorona97**
Added:
- Nonetype check for stochastic_ternary.alpha in __call__ method
- Test for stochastic_ternary.alpha = Nonetype behavior

### CODE REVIEW

**Comment by jecorona97** [comment]
@googlebot I signed it!

File: tests/qactivation_test.py [hunk 1]
```diff
@@ -417,34 +417,44 @@ def test_stochastic_binary():
...
+    ),
+    (
+        "auto_po2",
+        8, 
+        np.array([-0.998, -0.992, -0.992, -0.208,  0.048,  0.04 , 
+            0.448,  0.606, 0.987,  0.998]).astype(np.float32),
+        np.array([0.007812])
+    )
+])
+def test_stochastic_ternary(alpha, temperature, expected_values, expected_scale):
```
Comment (zhuangh) [inline tests/qactivation_test.py:436]: also add a case when alpha is not None and without substring "po2".

File: tests/qactivation_test.py [hunk 2]
```diff
@@ -417,34 +417,44 @@ def test_stochastic_binary():
   assert_allclose(scale, expected_scale, rtol=0.1)
 
 
-def test_stochastic_ternary():
+@pytest.mark.parametrize(
+    'alpha, temperature, expected_values, expected_scale', [
+    (
+        None, 
+        8,
+        np.array([0., 0., 0., 0., 0., 0., 0., 0., 0., 0.]).astype(np.float32),
```
Comment (zhuangh) [inline tests/qactivation_test.py:425]: case seems a little "easy", 
could you add more with -alpha, 0, +alpha, where alpha can be 1, or other numbers?
also different scales as well.

**Comment by zhuangh** [comment]
Hi Jose @jecorona97 

Thank you for this PR! comments added.

BTW, what about the example_act, sternary outputs? Are they expected to what you understand? I do not see the code change in this PR.

File: tests/qactivation_test.py (hunk 2 above)
Comment (jecorona97) [reply tests/qactivation_test.py:425]: Added these on the latest commit

File: tests/qactivation_test.py [hunk 3]
```diff
@@ -417,34 +417,65 @@ def test_stochastic_binary():
...
+    ),
+    (
+        0.0033,
+        8,
+        np.array([-1., -1., -0.999, -0.835,  0.474,  0.441,
+            0.949,  0.977,  0.999,  1.]).astype(np.float32),
+        np.array([0.0033])
+    ),
+    (
+        -0.0033,
```
Comment (zhuangh) [inline tests/qactivation_test.py:436]: could you change this alpha number to be larger than 1.

File: tests/qactivation_test.py (hunk 2 above)
Comment (zhuangh) [reply tests/qactivation_test.py:425]: thanks. to be more clear, I mean the value range between [-alpha, alpha].

File: qkeras/quantizers.py [hunk 4]
```diff
@@ -615,7 +615,7 @@ def __call__(self, x):
 
     m = K.max(tf.abs(x), axis=axis, keepdims=True)
     scale = 2.*m/3.
-    if "po2" in self.alpha:
+    if self.alpha is not None and "po2" in self.alpha:
```
Comment (zhuangh) [inline qkeras/quantizers.py:618]: also put an assertion that alpha should be not less than 0

**Comment by zhuangh** [comment]
pull to internal code review system.

**Pull request merged**